prov/tcp/src/tcpx_progress.c

```diff
@@ -383,10 +381,10 @@ static void tcpx_process_rx_msg(struct tcpx_ep *ep)
 	case TCPX_OP_REMOTE_WRITE:
 		process_rx_remote_write_entry(ep->cur_rx_entry);
 		break;
-	case TCPX_OP_READ:
+	case TCPX_OP_READ_RSP:
 		process_rx_read_entry(ep->cur_rx_entry);
 		break;
- 	case TCPX_OP_REMOTE_READ_REQ:
+ 	case TCPX_OP_REMOTE_READ:
```

Comment:
Note: You may be able to setup an array of function calls with op_data (validated) used as the index to make a direct call and avoid the switch statement.  This is just a suggestion for a future change.